assets/pages/contributions-landing/components/contributionsBundle.jsx

```diff
@@ -197,8 +198,8 @@ const getContribAttrs = (
   contribAmount: Amounts,
   referrerAcquisitionData: ReferrerAcquisitionData,
   isoCountry: IsoCountry,
+  countryGroupId: CountryGroupId,
   currency: IsoCurrency,
```

Comment:
There seem to be a few places where we're passing through some combination of `currency`, `country` and `countryGroup`, sometimes all three at once.

Could we unify this into a single piece of information that we can pass around? For example, in the `common` state these could exist as part of a single object, and if you ever find yourself needing 2 or more of these pieces of information, you pass through that object instead of sending each through individually?

Alternatively, as far as I understand the situation now, the only piece of information you need is the `country`, which can be used to derive the `countryGroup`, which can in turn be used to derive the `currency`. Is it possible that `country` is the only thing we need to pass through, and we can derive the others on-the-fly?

Reply:
> There seem to be a few places where we're passing through some combination of currency, country and countryGroup, sometimes all three at once.

Yes, that is the case because the in placed logic require some parts of it.

> Could we unify this into a single piece of information that we can pass around? For example, in the common state these could exist as part of a single object, and if you ever find yourself needing 2 or more of these pieces of information, you pass through that object instead of sending each through individually?

This is a great idea. Nevertheless, this implies a huge refactor. This PR is already too big.

> Alternatively, as far as I understand the situation now, the only piece of information you need is the country, which can be used to derive the countryGroup, which can, in turn, be used to derive the currency. Is it possible that country is the only thing we need to pass through, and we can derive the others on-the-fly?

Not always, the only piece which is 100% certain from the page you visit is the country group and sometimes the country (see EUR case). Then in the moment of a transaction, we ask for the concrete country (when we no know it). It is true that the currency can be computed from the countryGroup, but it will need an extra step everytime you want to use it. I would prefer to keep it inside the state.

Reply:
I created a card to do this refactor: https://trello.com/c/cftEZ0c8/1302-tidy-up-the-common-state